fix(quantic): update to breadcrumb style

nathanlb opened the pull request:
Updated breadcrumb component to fix long string truncating behaviour and swap the "X" for an icon.
To do this we added a `quanticPill` component to avoid duplicated markup for each breadcrumb.
Named it "Pill" over breadcrumb because it could be used in different contexts as well.

### Before
<img width="813" alt="Screen Shot 2021-10-04 at 9 03 43 AM" src="https://user-images.githubusercontent.com/16785453/135856267-1f11fcf0-6f05-428f-93a0-ced39fd06eaa.png">

### After

<img width="617" alt="Screen Shot 2021-10-04 at 9 04 34 AM" src="https://user-images.githubusercontent.com/16785453/135856411-53913819-db4d-44a6-8b9a-8dd06153fa4c.png">

<img width="617" alt="135872685-3fb221b8-c701-49cf-bb01-8c343ed808bd" src="https://user-images.githubusercontent.com/16785453/135898769-02ea43f6-a9e3-4b1b-af88-8cb72ea7ac67.png">

<img width="617" alt="Screen Shot 2021-10-04 at 9 05 31 AM" src="https://user-images.githubusercontent.com/16785453/135856507-64c96a08-f312-4c59-a1a6-9b835e7cecdf.png">

lbergeron commented on packages/quantic/force-app/main/default/lwc/quanticBreadcrumbManager/quanticBreadcrumbManager.html:
```diff
@@ -1,78 +1,78 @@
+              <ul class="slds-col slds-truncate slds-gutters slds-list_horizontal slds-grid slds-wrap">
+                <template for:each={breadcrumb.values} for:item="breadcrumbValue">
+                  <li key={breadcrumbValue.value.value} class="breadcrumb__container">
+                    <c-quantic-pill label={breadcrumbValue.value} alt-text={breadcrumbValue.value} deselect={breadcrumbValue.deselect}></c-quantic-pill>
```
Why are you using `breadcrumbValue.value` instead of `breadcrumbValue.value.value` like you do with regular facet values?

I feel the terminology around the breadcrumb levels is not that clear.

nathanlb replied:
Yeah it could use some refactoring. I tried not to touch the javascript on this one so this is the way it was before this update but i can give it a look.

nathanlb replied:
I looked into it and that's how the type is defined in headless. The breadcrumb.value has a different type based on the facet filter variant. Instead of adding logic in the component to map the breadcrumb values to some other format depending on the type I would prefer to take and display the values returned from headless as-is.

lbergeron commented on packages/quantic/force-app/main/default/lwc/quanticPill/quanticPill.js:
```diff
@@ -0,0 +1,27 @@
+   * @api
+   * @type {Function}
+   */
+  @api deselect;
```
I'm curious, what is the advantage of using a single callback function instead of dispatching an event?

What happens when the `deselect` property is not set? Does it break if a user clicks the pill?

nathanlb replied:
changed to use event

nathanlb commented:
> Do you think we should make the "Clear All Filters" displays in 1 line only (when the facetValue is really long)

I tried but i gave up after a few hours 😅  unless we give it a fixed width I can't get the flexbox to cooperative. Fixed width would work but then if a user modifies the label to say something else it might not be appropriate anymore.

lvu285 commented:
> > Do you think we should make the "Clear All Filters" displays in 1 line only (when the facetValue is really long)
> 
> I tried but i gave up after a few hours 😅 unless we give it a fixed width I can't get the flexbox to cooperative. Fixed width would work but then if a user modifies the label to say something else it might not be appropriate anymore.

but I prefer to have it works nice on our demo page, or maybe can you check with Audrey?

nathanlb commented:
> > > Do you think we should make the "Clear All Filters" displays in 1 line only (when the facetValue is really long)
> > 
> > 
> > I tried but i gave up after a few hours 😅 unless we give it a fixed width I can't get the flexbox to cooperative. Fixed width would work but then if a user modifies the label to say something else it might not be appropriate anymore.
> 
> but I prefer to have it works nice on our demo page, or maybe can you check with Audrey?

I fixed it!
Just needed to try again with a fresh look. 
<img width="617" alt="Screen Shot 2021-10-04 at 10 46 14 AM" src="https://user-images.githubusercontent.com/16785453/135872685-3fb221b8-c701-49cf-bb01-8c343ed808bd.png">
